webapp/src/components/boardTemplateSelector/boardTemplateSelectorPreview.tsx

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useEffect, useState, useMemo} from 'react'
+
+import {Board} from '../../blocks/board'
+import {Card} from '../../blocks/card'
+import {BoardView} from '../../blocks/boardView'
+import octoClient from '../../octoClient'
+import {getVisibleAndHiddenGroups} from '../../boardUtils'
+
+import ViewHeader from '../viewHeader/viewHeader'
+import ViewTitle from '../viewTitle'
+import Kanban from '../kanban/kanban'
+import Table from '../table/table'
+import CalendarFullView from '../calendar/fullCalendar'
+import Gallery from '../gallery/gallery'
+
+import './boardTemplateSelectorPreview.scss'
+
+type Props = {
+    activeTemplate: Board|null
+}
+
+const BoardTemplateSelectorPreview = React.memo((props: Props) => {
+    const {activeTemplate} = props
+    const [activeView, setActiveView] = useState<BoardView|null>(null)
+    const [activeTemplateCards, setActiveTemplateCards] = useState<Card[]>([])
+
+    useEffect(() => {
+        if (activeTemplate) {
+            setActiveTemplateCards([])
+            setActiveView(null)
+            setActiveTemplateCards([])
+            octoClient.getSubtree(activeTemplate.id, activeView?.fields.viewType === 'gallery' ? 3 : 2, activeTemplate.workspaceId).then((blocks) => {
+                const cards = blocks.filter((b) => b.type === 'card')
+                const views = blocks.filter((b) => b.type === 'view')
+                if (views.length > 0) {
+                    setActiveView(views[0] as BoardView)
+                }
+                if (cards.length > 0) {
+                    setActiveTemplateCards(cards as Card[])
+                }
+            })
+        }
+    }, [activeTemplate])
+
+    const dateDisplayProperty = useMemo(() => {
+        return activeTemplate?.fields.cardProperties.find((o) => o.id === activeView?.fields.dateDisplayPropertyId)
+    }, [activeView, activeTemplate])
+
+    const groupByProperty = useMemo(() => {
+        return activeTemplate?.fields.cardProperties.find((o) => o.id === activeView?.fields.groupById) || activeTemplate?.fields.cardProperties[0]
+    }, [activeView, activeTemplate])
+
+    const {visible: visibleGroups, hidden: hiddenGroups} = useMemo(() => {
+        if (!activeView) {
+            return {visible: [], hidden: []}
+        }
+        return getVisibleAndHiddenGroups(activeTemplateCards, activeView.fields.visibleOptionIds, activeView?.fields.hiddenOptionIds, groupByProperty)
+    }, [activeTemplateCards, activeView, groupByProperty])
+
+    if (!activeTemplate) {
+        return null
+    }
+
+    return (
+        <div className='BoardTemplateSelectorPreview'>
```

Comment:
Should this section be a separate component used both here and in `centerPanel.tsx`?

Reply:
I guess it can, but I would prefer to defer that until another PR. This is not strictly "repeated code", it is showing a preview, and the other one is showing the real thing. There is a lot of things populated with empty data here. Anyway, yes, we can make it less repeated code.

Reply:
I'm ok with that.